src/generic/grid.cpp

```diff
@@ -7770,7 +7719,20 @@ wxGrid::DoMoveCursorByBlock(bool expandSelection,
 
     if ( expandSelection )
     {
-        UpdateBlockBeingSelected(m_currentCellCoords, coords);
+        // TODO: Select the next block every time (not the same as now).
```

Comment:
I still have no idea about this one. I've refactored the code to fix the "TODO: pass keyboard state" part of this comment in the latest version, but what does the "same block" mean remains unclear.

Is this related to the currently wrong behaviour of `Shift-Ctrl-<Dir>`? Instead of extending the current block, it continues selection from the current cell, which is surprising and different from the standard spreadsheet behaviour.

Reply:
I've refactored the code to use the same initial cell coords in `DoMoveCursorByBlock()` as in `DoMoveCursor()` itself and this seems to fix the behaviour. Please let me know if you still something wrong from the user point of view here.